### PR TITLE
Improve coverage of some base agents

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Get up the code coverage of Starspace, IR baseline, and TFIDF a bit; as well as a few scripts.

A lot of deleting dead code.

**Testing steps**
New CI, codecov.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # .coveragerc to control coverage.py
 [run]
-branch = True
+branch = False
```
Comment (stephenroller) [inline .coveragerc:3]: this was intentional, it boosts our numbers 🤣

File: parlai/agents/tfidf_retriever/build_tfidf.py [hunk 2]
```diff
@@ -202,11 +108,8 @@ def get_count_matrix(args, db_opts):
     with DocDB(**db_opts) as doc_db:
         doc_ids = doc_db.get_doc_ids()
 
-    # Setup worker pool
     tok_class = tokenizers.get_class(args.tokenizer)
-    workers = ProcessPool(
-        args.num_workers, initializer=init, initargs=(tok_class, db_opts)
-    )
+    init(tok_class, db_opts)
```
[the remark this answers is not in the document]
Comment (stephenroller) [reply parlai/agents/tfidf_retriever/build_tfidf.py:112]: Yeah good question. So doing it in a pool speed things up, but breaks code coverage, so it all measures as 0.

We can debate this more.

File: tests/test_ir_baseline.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        valid, test = testing_utils.train_model(
+            {
+                'task': 'babi:task1k:1',
+                'model': 'ir_baseline',
+                'batchsize': 1,
+                'datatype': 'train:ordered',
+                'num_epochs': 1,
+            }
+        )
+        assert valid['f1'] == 0.41
```
[the remark this answers is not in the document]
Comment (stephenroller) [reply tests/test_ir_baseline.py:37]: assertEqual works better when you're using unittest, but pytest has some magic that makes the two equivalent. So I opt for the more readable version.

[the remark this answers is not in the document]
Comment (klshuster) [reply tests/test_ir_baseline.py:37]: oh awesome i'm gonna start doing that then

File: parlai/agents/tfidf_retriever/build_tfidf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/agents/tfidf_retriever/build_tfidf.py:112]: ok, my qualm is that the TFIDF retriever is best for larger datasets (at least in my experience), e.g. wikipedia. Not sure what type of slowdown we see without the pool but would be good to measure maybe.

why does using a pool break test coverage?

File: parlai/agents/ir_baseline/ir_baseline.py [hunk 4]
```diff
@@ -318,8 +310,6 @@ def act(self):
                 rep, cands, self.length_penalty, self.dictionary
             )
             reply['text'] = reply['text_candidates'][0]
-        else:
-            reply['text'] = "I don't know."
```
[the remark this answers is not in the document]
Comment (stephenroller) [reply parlai/agents/ir_baseline/ir_baseline.py:322]: I personally dislike this. Alex did it in a few places in some of our older models. I don't think the model should be making bits like this. I'd rather loudly throw an error.

File: parlai/agents/tfidf_retriever/build_tfidf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (stephenroller) [reply parlai/agents/tfidf_retriever/build_tfidf.py:112]: Maybe the right way is to use num workers to fall back to single threaded? Let’s do that instead

Idk exactly, but all the code executed in the pool didn’t hit coverage at all. It actually made deleting things sort of difficult— in an early version of this, I broke it totally when I deleted the uncovered code.

[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/agents/tfidf_retriever/build_tfidf.py:112]: yeah that works for me

File: parlai/agents/ir_baseline/ir_baseline.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/agents/ir_baseline/ir_baseline.py:322]: yeah totally agree, looks like you have tests covering this agent anyway so happy to have this

**Pull request merged**